src/main/java/com/JellyWorks/Utility/PreprocessFile.java

```diff
@@ -0,0 +1,109 @@
+package com.JellyWorks.Utility;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import com.JellyWorks.Utility.TypeChecks;
+import com.JellyWorks.storage.redis.RedisProcessor;
+
+import org.springframework.cache.annotation.Cacheable;
+import java.net.URISyntaxException;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.data.redis.core.StringRedisTemplate;
+import org.springframework.data.redis.core.ValueOperations;
+
+public class PreprocessFile {
+
+	private String path;
+	private InputStream inputStream;
+	private ConfigurableApplicationContext configurableApplicationContext;
+	
+	public PreprocessFile(InputStream i) throws IOException{
+		this.inputStream=i;
+		this.path="C:/Users/Ramya/Desktop/AWS/";
+		this.configurableApplicationContext = new AnnotationConfigApplicationContext(RedisProcessor.class);
+	}
+	@Cacheable("splitFile")
+	public boolean splitFile() throws URISyntaxException, Exception {
+		BufferedReader reader=new BufferedReader(new InputStreamReader(inputStream));
+		BufferedWriter writer=null;
+		try{
+			String line="";
+			int curr_month=0;
+			String out_folder="Output";
+			createDir(out_folder,path);
+			String outfile="Conversation_Month_";
+			String ext=".txt";
+			StringRedisTemplate redisTemplate = (StringRedisTemplate) configurableApplicationContext.getBean("strRedisTemplate");
+			ValueOperations<String, String> values = redisTemplate.opsForValue();
+			String key="default";
+			int count=0;
+			int month=0;
+		    while((line=reader.readLine()) != null){ 
+		    	//Need better logic to differentiate between message continuation. Regex char for split may be part of the message as well.
+		    	if(line.indexOf(",")!=-1){
+		    		String date_of_msg=line.substring(0, line.indexOf(","));
+		    		key=date_of_msg;
+		    		if(TypeChecks.isValidDate(date_of_msg)){
+		    			String month_of_msg=date_of_msg.substring(0,date_of_msg.indexOf("/"));
+			    		if(TypeChecks.isNumeric(month_of_msg))
+			    			month=Integer.parseInt(month_of_msg);
+				    	if(month!=curr_month){
+				    		if(writer!=null)
+				    			writer.close();
+				    		curr_month=month;
+				    		writer=new BufferedWriter(new FileWriter(path+out_folder+"/"+outfile+curr_month+ext));
+				    	}	
+		    		}
+		    	}
+		    	if(!writer.equals(null)){
+		    		//Need better logic for key
+		    		writer.write(line);
+		    		count++;
+		    		key=Integer.toString(count);
+		    		//if(values.get(key)!=null)
+		    			values.set(key, line);

```

Comment:
1. Redis reads and writes should be exposed by methods in RedisInvoker and RedisProcessor. 
2. The key is not meaningful. How are you going to query on the cache to obtain texts on a certain date or date range?
3. Each line of text makes a call to the cache. Currently it is a call to memory. Eventually when this is moved to an elastic cache, this will hog the network. Can you suggest a better way?

Reply:
Did you do something about this?